pom.xml

```diff
@@ -362,56 +362,38 @@
     </pluginManagement>
     <plugins>
       <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-shade-plugin</artifactId>
-        <version>3.0.0</version>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <version>3.5.1</version>
+        <extensions>true</extensions>
+        <configuration>
+          <instructions>
+            <Embed-Dependency>*;inline=false;scope=compile</Embed-Dependency>
+            <Embed-Transitive>true</Embed-Transitive>
+            <Embed-Directory>lib</Embed-Directory>
+            <!--Only @Plugin classes in the export packages will be included as plugin-->
+            <_exportcontents>
+              io.cdap.plugin.pubnub.*;
+            </_exportcontents>
+          </instructions>
+        </configuration>
         <executions>
           <execution>
             <phase>package</phase>
             <goals>
-              <goal>shade</goal>
+              <goal>bundle</goal>
             </goals>
-            <configuration>
-              <relocations>
-                <relocation>
-                  <pattern>com.google.gson</pattern>
-                  <shadedPattern>hidden.com.google.gson</shadedPattern>
```

Comment:
It seems like this was intentionally using shade instead of maven-bundle-plugin in order to rename gson

Have you tried running the plugins to make sure they still work after switching to maven-bundle-plugin?

Reply:
Yes I have tried running the plugin and it worked fine without shading.